AlgebraInLean/Chapter2/Sheet3.lean

```diff
@@ -0,0 +1,232 @@
+    to be the set of all endomorphisms of V, which we know to be nonempty
+    because of the existence of the endomorphism mapping some arbitrary vector
+    v ↦ 0, and the identity mapping v ↦ v.
+
+    -/
+
+  end Endomorphisms
+
+  section Automorphisms
+
+    /-
+
+    An automorphism is defined to be an endomorphism that is also a bijection.
+    You will recognize the following definition is similar to how we defined
+    bijectivity in the first place.
+
+    -/
+    def Automorphism [Group G] (φ : G → G) : Prop := Endomorphism φ ∧ Bijective φ
+
+    /-
+
+    You can think of it like a permutation from a group to itself, although it
+    is important that this permutation respects the group structure.
+    see more specifically what "respecting the group structure" looks like in
+    the next chapter (keep an eye out for orders!).
+
+    As an exercise, let's prove that a specific function mapping within the
+    group of integers under addition is a group automorphism.
+
+    Spefically, fix G = ⟨ℤ, +⟩, and φ : G → G, x ↦ -x
+    (the function fx = -x).
+
+    Note that in order to prove this, we do not necessarily need to "prove" that
+    our φ is an endomorphism. We are already defining it as the map φ : G → G
+    (a group onto itself), so it suffices to prove that φ is a homomorphism.
+    That may be useful going forward with this proof.
+
+    -/
+
+    -- A brief definition of our φ:
+    def φ (x : ℤ) : ℤ := -x
+    -- φ : G → G, x ↦ -x
+
+    -- Show that φ is a group automorphism
+    theorem φ_automorphism : ∀ x y : ℤ, φ (x + y) = φ x + φ y ∧ Bijective φ := by
+      intros x y
+      constructor
+      -- Prove homomorphism
+      · unfold φ
+        rw[neg_add]
+      -- Prove Bijectivity
+      · rw[Bijective]
+        constructor
+        -- Injectivity
+        · intros x y h
+          unfold φ at h
+          exact neg_inj.mp h
+        -- Surjectivity
+        · intro z
+          use -z
+          unfold φ
+          rw[neg_neg]
+      done
+
+
+    /-
+
+    Now that you have done a basic proof with automorphisms, we will move on to
+    one that is slightly more complex (which also introduces a new concept).
+    _conjugation_ is defined to be the specific relation between two elements of
+    some group where a,b ∈ G are _conjugates_ if there is also some g ∈ G such that
+    b = gag⁻¹
+
+    Specifically, in the case of the general linear group of invertible matrices,
+    _GL(n)_, this conjugacy relation is called matrix similarity, which may be
+    more familiar. (Recall that two matrices A,B are similar iff there is also some
+    matrix D such that B = DAD⁻¹).
+
+    We claim that conjugation is an automorphism under an arbitrary group and
+    group operation, meaning:
+    ψ : G → G, x ↦ gxg⁻¹ is a group automorphism. Let's prove this!
+
+    As with before, a brief definition of our ψ:
+
+    -/
+    def ψ [Group G] (g x : G) : G := μ (μ g x) (ι g)
```

Comment:
Rename psi to `conjugate`, and then I can change Chapter 3 to use that definition of `conjugate` instead of redefining it myself.

Reply:
Good idea -- done.